code/modules/mob/living/say.dm

```diff
@@ -250,8 +250,8 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 		return lowertext(message[1 + length(key)])
 
 /mob/living/proc/get_message_language(message)
-	if(message[1] == ",")
-		var/key = message[1 + length(message[1])]
+	if(length(message) >= 2 && message[1] == ",")
+		var/key = message[2]
```

Comment:
I think this was done this way because there could be two or more character radio/language codes.

Would you be so kind as to double check whether there are any in use?

Reply:
I did look at all the language keys and all are one character.

Would `length(message[1])` not always be **1** anyway though? If multiple char keys were intended then I think `length(LD.key)` should be used in the for loop to check.

Reply:
this is fine imo